charts/retool/templates/_workers.tpl

```diff
@@ -0,0 +1,413 @@
+                key: {{ $.Values.config.encryptionKeySecretKey | default "encryption-key" }}
+                {{- else }}
+                name: {{ template "retool.fullname" $ }}
+                key: encryption-key
+                {{- end }}
+          - name: POSTGRES_PASSWORD
+            valueFrom:
+              secretKeyRef:
+          {{- if  $.Values.postgresql.enabled }}
+                name: {{ template "retool.postgresql.fullname" $ }}
+                key: postgres-password
+          {{- else }}
+                {{- if $.Values.config.postgresql.passwordSecretName }}
+                name: {{ $.Values.config.postgresql.passwordSecretName }}
+                key: {{ $.Values.config.postgresql.passwordSecretKey | default "postgresql-password" }}
+                {{- else }}
+                name: {{ template "retool.fullname" $ }}
+                key: postgresql-password
+                {{- end }}
+          {{- end }}
+          - name: CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                {{- if $.Values.config.auth.google.clientSecretSecretName }}
+                name: {{ $.Values.config.auth.google.clientSecretSecretName }}
+                key: {{ $.Values.config.auth.google.clientSecretSecretKey | default "google-client-secret" }}
+                {{- else }}
+                name: {{ template "retool.fullname" $ }}
+                key: google-client-secret
+                {{- end }}
+          {{- end }}
+          {{- range $key, $value := $.Values.env }}
+          - name: "{{ $key }}"
+            value: "{{ $value }}"
+          {{- end }}
+          {{- range $.Values.environmentSecrets }}
+          - name: {{ .name }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ .secretKeyRef.name }}
+                key: {{ .secretKeyRef.key }}
+          {{- end }}
+          {{- with $.Values.environmentVariables }}
+{{ toYaml . | indent 10 }}
+          {{- end }}
+          {{- with $parentValues.config.environmentVariables }}
+{{ toYaml . | indent 10 }}
+          {{- end }}
+        {{- if $.Values.externalSecrets.enabled }}
+        envFrom:
+        - secretRef:
+            name: {{ $.Values.externalSecrets.name }}
+        {{- range $.Values.externalSecrets.secrets }}
+        - secretRef:
+            name: {{ .name }}
+        {{- end }}
+        {{- end }}
+        {{- if $.Values.externalSecrets.externalSecretsOperator.enabled  }}
+        envFrom:
+        {{- range $.Values.externalSecrets.externalSecretsOperator.secretRef }}
+        - secretRef:
+            name: {{ .name }}
+            optional: {{ .optional | default false }}
+        {{- end }}
+        {{- end }}
+        ports:
+        - containerPort: {{ $healthcheckPort }}
+          name: http-server
+          protocol: TCP
+        - containerPort: 9090
+          name: http-metrics
+          protocol: TCP
+
+{{- if $.Values.livenessProbe.enabled }}
```

Comment:
do all of these agents have readiness/liveness probes matching backend's? if not, you're going to get crashes when they try to come up.